<launch>

<!-- worldmap
	sel = 0 - Map with many files
	sel = 1 - Map with only 1 file
-->
	
<!-- DEMO
	sel = 0 - Make Map using RGB features
	sel = 1 - Make Map using RGB features and icp p2p
	sel = 2 - Make Map using RGB features and icp pl2pl
	sel = 3 - Make Map using RGB features and icp pl2pl vol
-->


<!-- -->	
	<!--node pkg="raposang_icp" name="world_node" type="world_node" -->
		<!--param name="sel" value="0" type="int" / -->	
		<!--param name="ply_filepath" value="/home/pedrovieira/Desktop/results/map_many_files/" type="str" / -->
		<!--param name="ply_filename" value="/home/pedrovieira/Desktop/results/cloud_world.ply" type="str" / -->
	<!--/node -->
<!-- -->		
		
		
	<node pkg="raposang_icp" name="demo_align_publisher" type="demo_align_publisher" output="screen">
		<param name="sel" value="0" type="int" />
		
		<!--ICP INIT-->
		<param name="th"   value="0.0001" type="double" />
		<param name="prob" value="0.999"  type="double" />
		
		<!--ICP-->
		<param name="D"					value="0.01" 	type="double" />		<!--5mm-->
		<param name="max_iterations" 	value="100" 	type="int"    />		
		<param name="THRESH_dr" 		value="0.5" 	type="double" />		<!--0.5 degrees-->
		<param name="THRESH_dt" 		value="0.002" 	type="double" />		<!--2mm meter 4mm para pl2pl-->
		<param name="MAXPAIRS" 			value="2000" 	type="int"    />		
		<param name="MINPAIRS"			value="200" 	type="int"    />
			
		<!--ICP pl2pl-->
		<param name="gicp_epsilon"		value="0.0004" 	type="double" />	
		<param name="k_correspondences" value="20" 		type="int"   />		

		<param name="pointcloudtopic"   value="/kinect/depth_camera/pointcloud_color_decompress" type="str" />	
	</node>
	
	
</launch>
